test/utils.py

```diff
@@ -260,25 +260,63 @@ def _save(self, p):
         self.outputs.append(p.outputs.copy())
 
 
-class ProcessSaver(plumpy.ProcessListener, Saver):
+_ProcessSaverProcReferences = {}
+_ProcessSaver_Saver = {}
+
+
+class ProcessSaver(plumpy.ProcessListener):
     """
     Save the instance state of a process each time it is about to enter a new state
     """
 
+    def __del__(self):
+        global _ProcessSaver_Saver
+        global _ProcessSaverProcReferences
+        if _ProcessSaverProcReferences is not None and id(self) in _ProcessSaverProcReferences:
+            del _ProcessSaverProcReferences[id(self)]
+        if _ProcessSaver_Saver is not None and id(self) in _ProcessSaver_Saver:
+            del _ProcessSaver_Saver[id(self)]
+
+    def get_process(self):
+        global _ProcessSaverProcReferences
+        return _ProcessSaverProcReferences[id(self)]
+
+    def _save(self, p):
+        global _ProcessSaver_Saver
+        _ProcessSaver_Saver[id(self)]._save(p)
+
+    def set_process(self, process):
+        global _ProcessSaverProcReferences
+        _ProcessSaverProcReferences[id(self)] = process
+
     def __init__(self, proc):
         plumpy.ProcessListener.__init__(self)
-        Saver.__init__(self)
-        self.process = proc
         proc.add_process_listener(self)
+        self.init_not_persistent(proc)
```

Comment:
Since `ProcessSaver` now just has a single direct base class, line 293 can be simplified from `plumpy.ProcessListener.__init__(self)` to `super.__init__()`